core/src/main/java/org/apache/iceberg/MetricsUtil.java

```diff
@@ -41,18 +41,36 @@ public class MetricsUtil {
   private MetricsUtil() {}
 
   /**
-   * Copies a metrics object without lower and upper bounds for given fields.
+   * Copies a metrics object without value, NULL and NaN counts for given fields.
    *
-   * @param excludedFieldIds field IDs for which the lower and upper bounds must be dropped
+   * @param excludedFieldIds field IDs for which the counts must be dropped
+   * @return a new metrics object without counts for given fields
+   */
+  public static Metrics copyWithoutFieldCounts(Metrics metrics, Set<Integer> excludedFieldIds) {
```

Comment:
I considered creating a builder for coping metrics but I am not sure it would be worth the extra complexity and it is not clear whether there would be future use cases benefiting from it. For now, I simply added another util method.